sefaria/helper/topic.py

```diff
@@ -107,6 +107,40 @@ def group_links_by_type(link_class, links, annotate_links, group_related):
                 grouped_links[link_type_slug]['pluralTitle'] = link_type.get('pluralDisplayName', is_inverse)
     return grouped_links
 
+def merge_props_for_similar_refs(curr_link, new_link):
+    # when grouping similar refs, make sure the source to display has the maximum curatedPrimacy of all the similar refs,
+    # as well as dataSource and descriptions of all the similar refs
+    if new_link.get('dataSource', None):
+        data_source = library.get_topic_data_source(new_link['dataSource'])
+        if data_source.slug == 'learning-team':
+            # use ref of learning team; in the case when both are learning team, use whichever ref covers a smaller range
+            if 'learning-team' not in curr_link['dataSources'] or len(curr_link['expandedRefs']) > len(new_link['expandedRefs']):
+                curr_link['ref'] = new_link['ref']
+                curr_link['expandedRefs'] = new_link['expandedRefs']
+        curr_link['dataSources'][new_link['dataSource']] = data_source.displayName
+        del new_link['dataSource']
+
+    if curr_link['is_sheet']:
+        return curr_link
+
+    new_description = new_link.get('descriptions', {})
+    if new_description != {}:
+        curr_link_description = curr_link.get('descriptions', {})
+        for lang in ['en', 'he']:
+            if lang not in curr_link_description and lang in new_description:
+                curr_link_description[lang] = new_description[lang]
+        curr_link['descriptions'] = curr_link_description
+
+    new_curated_primacy = new_link.get('order', {}).get('curatedPrimacy', {})
+    if new_curated_primacy != {}:
```

Comment:
if new_curated_primacy:
or
if not new_curated_primacy:
might be more readable here, and {} will return a bool in python